### PR TITLE
Disable signup endpoint option

### PR DESCRIPTION
**Pull request by TheoVRR**
- Added new config to disable signup endpoint
- Added new sign_up_disabled error

### CODE REVIEW

File: app/controllers/devise/api/tokens_controller.rb [hunk 1]
```diff
@@ -10,6 +10,13 @@ class TokensController < Devise.api.config.base_controller.constantize
 
       # rubocop:disable Metrics/AbcSize
       def sign_up
+        unless Devise.api.config.sign_up.enabled
```
Comment (nejdetkadir) [inline app/controllers/devise/api/tokens_controller.rb:13]:
```suggestion
        head :not_found unless Devise.api.config.sign_up.enabled
```

File: app/controllers/devise/api/tokens_controller.rb [hunk 2]
```diff
@@ -10,6 +10,13 @@ class TokensController < Devise.api.config.base_controller.constantize
...
+          error_response = Devise::Api::Responses::ErrorResponse.new(request, error: :sign_up_disabled,
+                                                                     resource_class: resource_class)
+
+          return render json: error_response.body, status: error_response.status
```
Comment (nejdetkadir) [inline app/controllers/devise/api/tokens_controller.rb:17]: I think we don't need to custom error mesage if registration disabled

Comment (TheoVRR) [reply app/controllers/devise/api/tokens_controller.rb:17]: I think it's not mandatory but it's more consistent with what was already done with the refresh_token_disabled error.

**Comment by GeorgesLeYeti** [comment]
Hi,

I'm very interested in this feature also. 

Any news about it ?

**Review by nejdetkadir — APPROVED** [review]

**Pull request merged**